master/internal/resourcemanagers/kubernetes_resource_manager.go

```diff
@@ -48,17 +50,12 @@ func (k *kubernetesResourceManager) Receive(ctx *actor.Context) error {
 	case actor.PreStart:
 		actors.NotifyAfter(ctx, actionCoolDown, schedulerTick{})
 
-	case sproto.ConfigureEndpoints:
-		ctx.Log().Infof("initializing endpoints for pods")
-		podsActor := kubernetes.Initialize(
-			msg.System,
-			msg.Echo,
-			ctx.Self(),
-			k.config.Namespace,
-			k.config.MasterServiceName,
-			k.config.LeaveKubernetesResources,
-		)
-		k.agent = newAgentState(sproto.AddAgent{Agent: podsActor})
+	case sproto.SetPods:
+		k.agent = &agentState{
```

Comment:
non-blocking: can we put check here that `k.agent` is not already configured since we should be setting it only once.

Reply:
added a panic check here.